csharp/AppEncryption/AppEncryption/Persistence/AdoMetastoreImpl.cs

```diff
@@ -37,17 +42,35 @@ public class AdoMetastoreImpl : IMetastore<JObject>
         private readonly string connectionString;
         private readonly DbProviderFactory dbProviderFactory;
 
-        public AdoMetastoreImpl(DbProviderFactory dbProviderFactory, string connectionString)
+        internal AdoMetastoreImpl(DbProviderFactory dbProviderFactory, string connectionString)
         {
             this.connectionString = connectionString;
             this.dbProviderFactory = dbProviderFactory;
         }
 
+        /// <summary>
+        /// A Builder method to create new instance of <see cref="AdoMetastoreImpl"/> class using the provided
+        /// parameters.
+        /// </summary>
+        ///
+        /// <param name="dbProviderFactory">The <seealso cref="dbProviderFactory"/> object which represents a set of
+        /// methods for creating instances of a provider's implementation of the data source classes.</param>
+        /// <param name="connectionString">The connection string to be used by the <seealso cref="DbConnection"/> in
+        /// <paramref name="dbProviderFactory"/>.</param>
```

Comment:
This can be rephrased. Too many links makes the param usage a little difficult to understand.